src/main/java/picard/analysis/artifacts/ContextAccumulator.java

```diff
@@ -41,7 +42,47 @@ public ContextAccumulator(final Set<String> contexts, final boolean expectedTand
     }
 
     public void countRecord(final String refContext, final char calledBase, final SAMRecord rec) {
-        artifactMap.get(refContext)[Transition.baseIndexMap[calledBase]].countRecord(rec);
+        AlignmentAccumulator[] accumulators = artifactMap.get(refContext);
+        if(accumulators != null) {accumulators[Transition.baseIndexMap[calledBase]].countRecord(rec);}
+    }
+
+    /**
+     * Fills a halfContextAccumualtor by summing over the appropriate counts from a fullContextAccumulator.
+     */
+    public void fillHalfRecords(final ContextAccumulator fullContextAccumulator, int contextSize){
+        final String padding = StringUtil.repeatCharNTimes('N', contextSize);
+        for(Map.Entry<String,AlignmentAccumulator[]> fullContext : fullContextAccumulator.artifactMap.entrySet()){
+            final String context = fullContext.getKey();
+            final char centralBase = context.charAt(contextSize);
+            final String leadingKey = context.substring(0, contextSize) + centralBase + padding;
+            final String trailingKey = padding + centralBase + context.substring(contextSize + 1, context.length());
+
+            AlignmentAccumulator[] trailingAlignmentAccumulators = this.artifactMap.get(trailingKey);
+            AlignmentAccumulator[] leadingAlignmentAccumulators = this.artifactMap.get(leadingKey);
+
+            for(int i=0; i < fullContext.getValue().length; i++){
```

Comment:
extract value into a final AlignmentAccumulator[] variable at the top of the loop.